pkgs/applications/audio/surge/default.nix

```
@@ -0,0 +1,49 @@
+{ stdenv, fetchFromGitHub, premake5, pkgconfig, cmake
+,cairo, libxkbcommon, libxcb, xcb-util-cursor, xcbutilkeysyms, ncurses, which, getopt
+,python, gnome3, lato }:
+
+stdenv.mkDerivation rec {
+  pname = "surge";
+  version = "1.6.6";
+
+  src = fetchFromGitHub {
+    owner = "surge-synthesizer";
+    repo = pname;
+    rev = "release_${version}";
+    sha256 = "0al021f516ybhnp3lhqx8i6c6hpfaw3gqfwwxx3lx3hh4b8kjfjw";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ premake5 pkgconfig cmake ];
+
+  buildInputs = [  cairo libxkbcommon libxcb xcb-util-cursor xcbutilkeysyms ncurses which getopt python gnome3.zenity lato ];
+
+  patchPhase = ''
+    patchShebangs build-linux.sh
+    substituteInPlace src/common/SurgeStorage.cpp --replace "/usr/share/Surge" "$out/share/surge"
+    cat src/common/SurgeStorage.cpp
+  '';
+
+  buildPhase = ''
+      ./build-linux.sh build
+    '';
```

```suggestion
  configurePhase = ''
    ./build-linux.sh premake
    python scripts/linux/emit-vector-piggy.py .
  '';

  buildFlags = [ "config=release_x64" ];

  enableParallelBuilding = true;
```
We can enable parallel building if we do the steps from `build-linux.sh build` ourselves. The normal build process is single threaded, so this is much faster on my machine :)

---

That doesn't build anything here:
```
replacing old 'surge-1.6.6'
installing 'surge-1.6.6'
these derivations will be built:
  /nix/store/65zqvpm63hggg2a3mrvvwrrlracildfh-surge-1.6.6.drv
building '/nix/store/65zqvpm63hggg2a3mrvvwrrlracildfh-surge-1.6.6.drv' on 'ssh://nxb-4'...
Warning: Permanently added 'beta.nixbuild.net,148.251.182.157' (ED25519) to the list of known hosts.
16083: start build
principal> starting
fs> mounting /nix/store-ro
agent> boot time: 0.406580
agent> executing /nix/bin/nix-store
unpacking sources
unpacking source archive /nix/store/1342bc1fh5y0jh743chq9mbwafdkl30p-source
source root is source
patching sources
patching script interpreter paths in build-linux.sh
build-linux.sh: interpreter directive changed from "/bin/bash" to "/nix/store/z4ajipns0l1s8b2lrgpy6nng4cys7h99-bash-4.4-p23/bin/bash"
configuring
configure flags: gmake
Building configurations...
Running action 'gmake'...
Generated Makefile...
Generated surge-vst3.make...
Generated surge-lv2.make...
Done (120ms).
building
Building configurations...
Running action 'gmake2'...
Generated Makefile...
Generated surge-vst3.make...
Generated surge-lv2.make...
Done (418ms).
glibPreInstallPhase
installing
cp: cannot stat 'target/headless/Release/Surge/Surge-Headless': No such file or directory
builder for '/nix/store/65zqvpm63hggg2a3mrvvwrrlracildfh-surge-1.6.6.drv' failed with exit code 1
16083: failed: NixPermanentFailure builder for '/nix/store/65zqvpm63hggg2a3mrvvwrrlracildfh-surge-1.6.6.drv' failed with exit code 1
error: build of '/nix/store/65zqvpm63hggg2a3mrvvwrrlracildfh-surge-1.6.6.drv' on 'ssh://nxb-4' failed: builder for '/nix/store/65zqvpm63hggg2a3mrvvwrrlracildfh-surge-1.6.6.drv' failed with exit code 1
builder for '/nix/store/65zqvpm63hggg2a3mrvvwrrlracildfh-surge-1.6.6.drv' failed with exit code 1
error: build of '/nix/store/65zqvpm63hggg2a3mrvvwrrlracildfh-surge-1.6.6.drv' failed
```

---

That doesn't fully build here:
```
x11platform.cpp
x11timer.cpp
x11utils.cpp
Linking surge-vst3
Running postbuild commands
./scripts/linux/package-vst3.sh
Creating Linux VST3 Bundle...
Linking surge-lv2
Running postbuild commands
python scripts/linux/generate-lv2-ttl.py target/lv2/Release/Surge.lv2/Surge.so
glibPreInstallPhase
installing
cp: cannot stat 'target/headless/Release/Surge/Surge-Headless': No such file or directory
```

---

Right, it doesn't install `Surge-Headless`. Sorry, I wrote a comment for that, but it looks like I forgot to add it :/

Here goes: When I run `Surge-Headless`, I get
```
$ ./Surge-Headless
surge-headless is our regtest engine for the synth engine excluding the UI
```
and a bunch of errors. Is there a reason to include it in the release? My feeling is it could be run in a `checkPhase`, to see if the build went well, but I couln't make that work :)

---

Headless requires an install of shared assets and also the test data in the build did. If run from the build dir after assets are installed it should give 100% passes. Without knowing what the errors were it is hard for me to offer more 

You can check out the azure-pipelines.yml to see how we run headless in ci. It runs on every pr and merge and has to pass in that setting.

HTH

---

@baconpaul Thanks, that sure helps!
I found out that at least one issue is that it can't find it's WAV's, since it's not running from the build dir: NixOS installs every program in it's own dir. 
 
@puzzlewolf I also tried a bit more to get it working, and will give up for now.
I'll push your changes asap.

For completeness sake, here's the errors I get:

When I run `` Surge-Headless > surge.txt `` I get this:
```
Surge Error
Wavetable Import Error
Unable to open file 'test-data/wav/Wavetable.wav'
Surge Error
Wavetable Import Error
Unable to open file 'test-data/wav/05_BELL.WAV'
Surge Error
Wavetable Import Error
Unable to open file 'test-data/wav/pluckalgo.wav'
```
Here's surge.txt:
```


surge-headless is our regtest engine for the synth engine excluding the UI
Run 'surge-headless --help' for options.


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Surge-Headless is a Catch v2.9.2 host application.
Run with -? for options

-------------------------------------------------------------------------------
Unison Absolute and Relative
  Wavetable Oscillator
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:110
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:66: FAILED:
  REQUIRE( surge->loadPatchByPath( pn, -1, "Test" ) )
with expansion:
  false

-------------------------------------------------------------------------------
Unison Absolute and Relative
  Window Oscillator
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:116
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:66: FAILED:
  REQUIRE( surge->loadPatchByPath( pn, -1, "Test" ) )
with expansion:
  false

-------------------------------------------------------------------------------
Unison Absolute and Relative
  Classic Oscillator
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:122
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:66: FAILED:
  REQUIRE( surge->loadPatchByPath( pn, -1, "Test" ) )
with expansion:
  false

-------------------------------------------------------------------------------
Unison Absolute and Relative
  SH Oscillator
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:128
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:66: FAILED:
  REQUIRE( surge->loadPatchByPath( pn, -1, "Test" ) )
with expansion:
  false

-------------------------------------------------------------------------------
Unison at Sample Rates
  Wavetable Oscillator
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:211
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:139: FAILED:
  REQUIRE( surge->loadPatchByPath( pn, -1, "Test" ) )
with expansion:
  false
with message:
  Wavetable test at 44100

-------------------------------------------------------------------------------
Unison at Sample Rates
  Window Oscillator
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:226
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:139: FAILED:
  REQUIRE( surge->loadPatchByPath( pn, -1, "Test" ) )
with expansion:
  false
with message:
  Window test at 44100

-------------------------------------------------------------------------------
Unison at Sample Rates
  Classic Oscillator
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:239
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:139: FAILED:
  REQUIRE( surge->loadPatchByPath( pn, -1, "Test" ) )
with expansion:
  false
with message:
  Classic test at 44100

-------------------------------------------------------------------------------
Unison at Sample Rates
  SH Oscillator
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:252
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsDSP.cpp:139: FAILED:
  REQUIRE( surge->loadPatchByPath( pn, -1, "Test" ) )
with expansion:
  false
with message:
  SH test at 44100

-------------------------------------------------------------------------------
We can read a collection of wavetables
  Wavetable.wav
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:31
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:35: FAILED:
  REQUIRE( wt->size == 2048 )
with expansion:
  1024 (0x400) == 2048 (0x800)

-------------------------------------------------------------------------------
We can read a collection of wavetables
  05_BELL.WAV
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:40
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:44: FAILED:
  REQUIRE( wt->size == 2048 )
with expansion:
  1024 (0x400) == 2048 (0x800)

-------------------------------------------------------------------------------
We can read a collection of wavetables
  pluckalgo.wav
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:49
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:53: FAILED:
  REQUIRE( wt->size == 2048 )
with expansion:
  1024 (0x400) == 2048 (0x800)

-------------------------------------------------------------------------------
DAW Streaming and Unstreaming
  SCL State Saves
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:178
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:188: FAILED:
  REQUIRE( !surgeSrc->storage.isStandardTuning )
with expansion:
  false

-------------------------------------------------------------------------------
DAW Streaming and Unstreaming
  Save and Restore KBM
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:203
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:214: FAILED:
  REQUIRE( !surgeSrc->storage.isStandardMapping )
with expansion:
  false

-------------------------------------------------------------------------------
Stream WaveTable Names
  Name Restored for Old Patch
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:239
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:243: FAILED:
  REQUIRE( surge->loadPatchByPath( "test-data/patches/Church.fxp", -1, "Test" ) )
with expansion:
  false

-------------------------------------------------------------------------------
Load Patches with Embedded KBM
  Check Restore
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:342
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:347: FAILED:
  REQUIRE( !surge->storage.isStandardTuning )
with expansion:
  false

-------------------------------------------------------------------------------
IDs are Stable
  Compare IDs
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:396
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsIO.cpp:450: FAILED:
  REQUIRE( nameIdMap.find( sn ) != nameIdMap.end() )
with expansion:
  {?} != {?}
with messages:
  Comparing current surge with test-data/param-ids-1.6.5.txt
  Comparing volume_FX1 with id 0

-------------------------------------------------------------------------------
Channel Split Routes on Channel
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsMIDI.cpp:16
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsMIDI.cpp:20: FAILED:
  REQUIRE( surge->loadPatchByPath( "test-data/patches/ChannelSplit-Sin-2OctaveB.fxp", -1, "Test" ) )
with expansion:
  false

-------------------------------------------------------------------------------
Retune Surge to .scl files
  Zeus 22
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:32
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:37: FAILED:
  REQUIRE( n2f(surge->storage.scaleConstantNote()+s.count) == surge->storage.scaleConstantPitch()*2 )
with expansion:
  32.0f == 64.0f

-------------------------------------------------------------------------------
Retune Surge to .scl files
  6 exact
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:42
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:47: FAILED:
  REQUIRE( n2f(surge->storage.scaleConstantNote()+s.count) == surge->storage.scaleConstantPitch()*2 )
with expansion:
  32.0f == 64.0f

-------------------------------------------------------------------------------
Notes at Appropriate Frequencies
  Zeus 22
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:92
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:101: FAILED:
  REQUIRE( fDouble == Approx( 261.63 * 2 ).margin( .1 ) )
with expansion:
  261.6314697266 == Approx( 523.26 )

-------------------------------------------------------------------------------
Notes at Appropriate Frequencies
  6 exact
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:105
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:115: FAILED:
  REQUIRE( fDouble == Approx( 261.63 * 2 ).margin( .1 ) )
with expansion:
  261.6314697266 == Approx( 523.26 )

-------------------------------------------------------------------------------
KBM File Parsing
  Parse A440 File
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:141
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:144: FAILED:
  REQUIRE( k.name == "test-data/scl/mapping-a440-constant.kbm" )
with expansion:
  ""
  ==
  "test-data/scl/mapping-a440-constant.kbm"

-------------------------------------------------------------------------------
KBM File Parsing
  Parse 7 to 12 mapping File
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:158
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:161: FAILED:
  REQUIRE( k.name == "test-data/scl/mapping-a442-7-to-12.kbm" )
with expansion:
  ""
  ==
  "test-data/scl/mapping-a442-7-to-12.kbm"

-------------------------------------------------------------------------------
KBM File Remaps Center
  Marvel 12 unmapped
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:184
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:194: FAILED:
  REQUIRE( f69 == Approx( 448.2 ).margin( .1 ) )
with expansion:
  439.9971618652 == Approx( 448.2 )

-------------------------------------------------------------------------------
KBM File Remaps Center
  And remap to 440
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:200
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:212: FAILED:
  REQUIRE( unmapped[2]/440.0 == Approx( unmapped[0] / f60 ).margin(.001) )
with expansion:
  nan == Approx( 0.0 )

-------------------------------------------------------------------------------
Non-uniform keyboard mapping
  7 Note Scale
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:315
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:328: FAILED:
  REQUIRE( ! k.isStandardMapping )
with expansion:
  false

-------------------------------------------------------------------------------
Non-uniform keyboard mapping
  7 Note Scale with Tuning Centers
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:348
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:352: FAILED:
  REQUIRE( ! k261.isStandardMapping )
with expansion:
  false

-------------------------------------------------------------------------------
Zero Size Maps
  Note 61
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:380
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:387: FAILED:
  REQUIRE( !k61.isStandardMapping )
with expansion:
  false

-------------------------------------------------------------------------------
Zero Size Maps
  Note 69
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:397
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:405: FAILED:
  REQUIRE( !k69.isStandardMapping )
with expansion:
  false

-------------------------------------------------------------------------------
Zero Size Maps
  Note 69 with Tuning
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:415
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:422: FAILED:
  REQUIRE( f69std != Approx( 440.0 ).margin( 0.1 ) )
with expansion:
  439.9971618652 != Approx( 440.0 )

-------------------------------------------------------------------------------
An Octave is an Octave
  22 note scale OSC Octave
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:506
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:510: FAILED:
  REQUIRE( s.count == 22 )
with expansion:
  0 == 22

-------------------------------------------------------------------------------
An Octave is an Octave
  22 note scale Scene Octave
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:524
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:528: FAILED:
  REQUIRE( s.count == 22 )
with expansion:
  0 == 22

-------------------------------------------------------------------------------
An Octave is an Octave
  6 note scale OSC Octave
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:542
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:546: FAILED:
  REQUIRE( s.count == 6 )
with expansion:
  0 == 6

-------------------------------------------------------------------------------
An Octave is an Octave
  6 note scale Scene Octave
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:560
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:564: FAILED:
  REQUIRE( s.count == 6 )
with expansion:
  0 == 6

-------------------------------------------------------------------------------
Non-Monotonic Tunings
  SCL Non-monotonicity
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:582
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:602: FAILED:
  REQUIRE( straight[i] == Approx( shuffle[indices[i]] ).margin( 0.1 ) )
with expansion:
  277.17709f == Approx( 293.6653137207 )
with message:
  Comparing index 1 with 2

-------------------------------------------------------------------------------
Non-Monotonic Tunings
  KBM Non-monotonicity
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:606
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:623: FAILED:
  REQUIRE( straight[i] == Approx( shuffle[indices[i]] ).margin( 0.1 ) )
with expansion:
  277.17709f == Approx( 293.6653137207 )
with message:
  Comparing index 1 with 2

-------------------------------------------------------------------------------
Mapping below and outside of count
  A bit below
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:630
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:638: FAILED:
  REQUIRE( surge->storage.table_pitch[54+256] == Approx( 259.6 / 8.175798915 ).margin( 1e-4 ) )
with expansion:
  22.62742f == Approx( 31.7522486425 )

-------------------------------------------------------------------------------
Mapping below and outside of count
  Twelve below
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:645
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:653: FAILED:
  REQUIRE( surge->storage.table_pitch[48+256] == Approx( 100.0 / 8.175798915 ).margin( 1e-4 ) )
with expansion:
  16.0f == Approx( 12.2312205865 )

-------------------------------------------------------------------------------
Mapping below and outside of count
  A lot below
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:660
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:668: FAILED:
  REQUIRE( surge->storage.table_pitch[42+256] == Approx( 100.0 / 8.175798915 ).margin( 1e-4 ) )
with expansion:
  11.31371f == Approx( 12.2312205865 )

-------------------------------------------------------------------------------
Mapping below and outside of count
  Twelve above
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:675
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:683: FAILED:
  REQUIRE( surge->storage.table_pitch[72+256] == Approx( 500.0 / 8.175798915 ).margin( 1e-4 ) )
with expansion:
  64.0f == Approx( 61.1561029324 )

-------------------------------------------------------------------------------
Mapping below and outside of count
  A lot above
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:690
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:698: FAILED:
  REQUIRE( surge->storage.table_pitch[80+256] == Approx( 1000.0 / 8.175798915 ).margin( 1e-4 ) )
with expansion:
  101.59369f == Approx( 122.3122058647 )

-------------------------------------------------------------------------------
Mapping below and outside of count
  A bit below with 6ns
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:705
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:716: FAILED:
  REQUIRE( surge->storage.table_pitch[54+256] == Approx( 259.6 / 8.175798915 ).margin( 1e-4 ) )
with expansion:
  22.62742f == Approx( 31.7522486425 )

-------------------------------------------------------------------------------
Mapping below and outside of count
  A lot below witn 6ns
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:723
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:734: FAILED:
  REQUIRE( surge->storage.table_pitch[42+256] == Approx( 100.0 / 8.175798915 ).margin( 1e-4 ) )
with expansion:
  11.31371f == Approx( 12.2312205865 )

-------------------------------------------------------------------------------
Mapping below and outside of count
  A lot above with 6ns
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:741
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:752: FAILED:
  REQUIRE( surge->storage.table_pitch[80+256] == Approx( 1000.0 / 8.175798915 ).margin( 1e-4 ) )
with expansion:
  101.59369f == Approx( 122.3122058647 )

-------------------------------------------------------------------------------
Mapping below and outside of count
  A lot below witn ED3-17
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:759
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:770: FAILED:
  REQUIRE( surge->storage.table_pitch[42+256] == Approx( 100.0 / 8.175798915 ).margin( 1e-4 ) )
with expansion:
  11.31371f == Approx( 12.2312205865 )

-------------------------------------------------------------------------------
Mapping below and outside of count
  A lot above with ED3-17
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:777
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:788: FAILED:
  REQUIRE( surge->storage.table_pitch[80+256] == Approx( 1000.0 / 8.175798915 ).margin( 1e-4 ) )
with expansion:
  101.59369f == Approx( 122.3122058647 )

-------------------------------------------------------------------------------
Ignoring Tuning Tables are Correct
  Ignoring in retuned compares with untuned
-------------------------------------------------------------------------------
/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:893
...............................................................................

/tmp/nix-build-surge-1.6.6.drv-0/source/src/headless/UnitTestsTUN.cpp:904: FAILED:
  REQUIRE( surge->storage.note_to_pitch(e) != surgeTuned->storage.note_to_pitch(e) )
with expansion:
  1508701.0f != 1508701.0f
with message:
  Testing condition reported in #1545

===============================================================================
test cases:     28 |     10 passed | 18 failed
assertions: 475918 | 475871 passed | 47 failed
```

---

@puzzlewolf OK, all done.

---

Right so that makes perfect sense. Basically you need to make the test-data available wherever you run it. So either (1) symlink or copy the test-data directory to where you run it or (2) push an upstream change to the tests where you allow an environment variable to repoint that directory (which I would happily merge) or (3) don’t run the tests at install. I like 1 or 2, but can also understand 3. the tests are relatively new (I started writing them after 1.6.4 when we just kept hitting irritating tuning and dsp bugs) so only our automatic builds run them at this time; KX and the various arches do not.

Best